frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx

```diff
@@ -59,50 +83,65 @@ const TemplatesTableLogic: FC<ITemplateTableLogicProps> = ({ ...props }) => {
             data,
           } = subscriptionData as UpdatedOwnedInstancesSubscriptionResult;
 
-          if (!data?.updateInstance?.instance) return prev;
-
-          const { instance } = data?.updateInstance;
+          const { instance } = data?.updateInstance!;
+          let { updateType } = data?.updateInstance!;
 
           if (prev.instanceList?.instances) {
-            let { instances } = prev.instanceList;
-            if (
-              instances.find(i => i?.metadata?.name === instance.metadata?.name)
-            ) {
-              instances = instances.map(i =>
-                i?.metadata?.name === instance.metadata?.name ? instance : i
+            if (updateType === UpdateType.Deleted) {
+              setDataInstances(old =>
+                old?.filter(i => i?.idName !== instance!.metadata?.name)
+              );
+            } else if (updateType === UpdateType.Modified) {
+              setDataInstances(old =>
+                old?.map((i, n) =>
+                  i.idName === instance?.metadata?.name!
+                    ? getInstance(instance!, n)
+                    : i
+                )
```

Comment:
```suggestion
                Object.assign(old.find(({ name }) => instance?.metadata?.name! === name), getInstance(instance!))
```
What about an approach of this type? By not getting an id, getInstance will not set it and Object.assign should not overwrite the old one.